src/components/table/column-picker/column-picker.component.html

```diff
@@ -12,39 +12,84 @@
         </ng-container>
     </div>
 
-    <div class="column-picker-list" [(uxSelection)]="_deselectedSelection">
+    <cdk-tree class="column-picker-list"
+              [dataSource]="_treeDataSource"
+              [treeControl]="_treeControl"
+              [(uxSelection)]="_deselectedSelection"
+              uxTabbableList>
+
+        <!-- Create item for not expandable node -->
+        <cdk-tree-node *cdkTreeNodeDef="let node">
+            <div uxTabbableListItem
+                 [uxSelectionItem]="node"
+                 *ngIf="_shouldRenderNode(node)"
+                 class="column-picker-list-item column-picker-tree-node-level-{{ node.level }}">
+
+                <ng-container *ngIf="!deselectedTemplate">{{ node.name }}</ng-container>
+
+                <ng-container
+                    *ngIf="deselectedTemplate"
+                    [ngTemplateOutlet]="deselectedTemplate"
+                    [ngTemplateOutletContext]="{ $implicit: node.name }">
+                </ng-container>
+            </div>
 
-        <div *ngFor="let column of deselected"
-             class="column-picker-list-item"
-             [uxSelectionItem]="column">
+        </cdk-tree-node>
 
-            <ng-container *ngIf="!deselectedTemplate">{{ column }}</ng-container>
+        <!-- Create item for expandable node -->
+        <cdk-tree-node *cdkTreeNodeDef="let node; when: _nodeHasChildren">
 
-            <ng-container
-                *ngIf="deselectedTemplate"
-                [ngTemplateOutlet]="deselectedTemplate"
-                [ngTemplateOutletContext]="{ $implicit: column }">
-            </ng-container>
-        </div>
-    </div>
+            <div *ngIf="_nodeHasAvailableChildren(node)"
+                 class="column-picker-tree-group-node">
+
+                <button uxTabbableListItem
+                        (click)="_setNodeExpanded(node, !node.isExpanded)"
+                        (keydown.arrowright)="_setNodeExpanded(node, true)"
+                        (keydown.arrowleft)="_setNodeExpanded(node, false)"
+                        [attr.aria-label]="'Toggle ' + node.name"
```

Comment:
If you are applying an aria-label within a component then we need to expose a localization hook. However, I believe if you apply the correct roles and aria attributes, and use the appropriate keyboard conventions, the label may not be necessary. Also this button isn't part of the tab ring, so this label will not be encountered.

I suggest reviewing this ARIA practices document and making sure this matches as closely as possible:
https://www.w3.org/TR/wai-aria-practices/examples/treeview/treeview-2/treeview-2a.html

If you haven't already, install ChromeVox for testing.

Reply:
Sorry I just copied and pasted their example, didn't give this enough thought so I need to be mindful in future. I'll get it fixed to match the guildlines.

Reply:
I've went through the guidelines and it will not be possible to follow it fully - we would need to use the CdkNestedTree to follow it fully and that will also lead to more issues with keyboard navigation. The only part missing is the ability to set `role="group"` on the child items. It will mean that the audio won't report that a group is anymore than a button. I implemented a overridable function in the same way that the selected reorder has been implemented so that it is clear the group items are collapsable / expandable.

Reply:
If you check the DOM you can see the `role` has already been set for you by the CDK tree, so that's not a problem.